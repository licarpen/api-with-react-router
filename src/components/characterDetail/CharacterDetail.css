.CharacterDetail{
  font-family: Arial, Helvetica, sans-serif;
  margin: auto;
  text-align: center;

  li{
    list-style: none;
  }
}