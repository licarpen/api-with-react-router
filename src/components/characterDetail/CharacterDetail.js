import React from 'react';
import PropTypes from 'prop-types';
import { useCharacter } from '../../hooks/charactersList';

const CharacterDetail = ({ match }) => {
  const { selectedCharacter } = useCharacter(match.params.id);

  return (
    <div>
      <h1>{selectedCharacter.name}</h1>
    </div>
  );
};

CharacterDetail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number.isRequired
    }).isRequired
  }).isRequired
};

export default CharacterDetail;
