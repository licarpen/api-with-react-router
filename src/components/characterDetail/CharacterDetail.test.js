import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<CharacterDetail match={{ params: {
      id: '1'
    } }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
