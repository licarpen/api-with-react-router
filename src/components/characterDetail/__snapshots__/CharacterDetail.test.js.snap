// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component matches a snapshot 1`] = `
<div>
  <h1 />
</div>
`;
